web/Controllers/PropertyController.cs

```diff
@@ -123,4 +124,18 @@ public ActionResult AddNewProperty_Cancel()
         // Go to View Properties page
         return RedirectToAction("ViewProperties", "Landlord");
     }
-}
+
+    [HttpGet]
+    [Route("/property/{propertyId:int}/view")]
+    public ActionResult ViewProperty(int propertyId)
+    {
+        PropertyDbModel? model = _propertyService.GetPropertyByPropertyId(propertyId);
+        if (model == null)
+        {
+            _logger.LogWarning("Property with ID {PropertyId} does not exist", propertyId);
+            return StatusCode(404);

```

Comment:
You could use the `Error` endpoint in the `HomeController` to make use of the nicer error pages Glenn and Claire created by redirecting to this action, something like:
```
return RedirectToAction("Error", "Home", new { status = 404 });
```

Reply:
ok sure done :-)